grammars/php.cson

```diff
@@ -3724,3 +3733,26 @@
         ]
       }
     ]
+  'ternary_shorthand':
+    'match': '\\?:'
+    'name': 'keyword.operator.ternary.php'
+  'ternary_expression':
+    'begin': '(?<!\\?)\\s*(\\?)\\s*(?!:|\\?)'
+    'beginCaptures':
+      '1':
+        'name': 'keyword.operator.ternary.php'
+    'end': '\\s*(:)\\s*'
+    'endCaptures':
+      '1':
+        'name': 'keyword.operator.ternary.php'
+    'patterns': [
+      {
+        'include': '#language'
+      }
+      {
+        'include': '$self'
```

Comment:
What's this `$self` match doing here?

Reply:
Inside a ternary can be any valid php expression. Since there is no "expression" defined, I just reinclude the entire language so that everything is available. 

Is this not to correct way to do that?

Reply:
Oh whoops, I misstated that.. I think I put the self there along with language because ternaries can be nested.

Reply:
Only `#language` should be included as `$self` would include the HTML grammar instead: https://github.com/atom/language-php/blob/aa095129f29f4ffbf5db6f154aa283997ddbca29/grammars/php.cson#L135

Reply:
Ah.. I think my confusion was that I thought `$self` was scoped at or below the current scope level and not the entire grammar. 

Thanks for the clarification.

Reply:
`$self` includes everything in the top-level `patterns` block.

Reply:
Got it. Fixed.